tools-and-tests/tools/src/main/java/org/hiero/block/tools/days/subcommands/GcpLsCommand.java

```diff
@@ -0,0 +1,399 @@
+            defaultValue = "10")
+    private int samples;
+
+    @Option(names = "--start-date", required = true, description = "Start date (inclusive), yyyy-MM-dd")
+    private LocalDate startDate;
+
+    @Option(names = "--end-date", required = true, description = "End date (inclusive), yyyy-MM-dd")
+    private LocalDate endDate;
+
+    @Option(
+            names = "--downloaded-days-dir",
+            description = "Root of downloadedDays directory (YYYY/MM/DD/record0.0.<node> structure)")
+    private Path downloadedDaysDir;
+
+    @Option(
+            names = "--node-count",
+            description = "Number of record nodes (record0.0.x), usually 31 for 0..30",
+            defaultValue = "31")
+    private int nodeCount;
+
+    @Option(
+            names = "--skip-local-compare",
+            description = "Skip comparing against local downloadedDays dir; only show GCP signature counts",
+            defaultValue = "false")
+    private boolean skipLocalCompare;
+
+    @Option(
+            names = "--show-ls",
+            description = "Show full list of GCP/local record files per node/hour",
+            defaultValue = "false")
+    private boolean showLs;
+
+    @Option(
+            names = {"-c", "--cache-dir"},
+            description = "Directory for GCS cache (default: data/gcp-cache)",
+            defaultValue = "data/gcp-cache")
+    private File cacheDir;
+
+    @Option(
+            names = {"--cache"},
+            description = "Enable GCS caching (default: false)",
+            defaultValue = "false")
+    private boolean cacheEnabled;
+
+    @Option(
+            names = {"--min-node"},
+            description = "Minimum node account ID (default: 3)",
+            defaultValue = "3")
+    private int minNodeAccountId;
+
+    @Option(
+            names = {"--max-node"},
+            description = "Maximum node account ID (default: 37)",
+            defaultValue = "37")
+    private int maxNodeAccountId;
+
+    @Option(names = "--parallelism", description = "Thread pool size for parallel node queries", defaultValue = "8")
+    private int parallelism;
+
+    @Option(
+            names = {"-p", "--user-project"},
+            description = "GCP project to bill for requester-pays bucket access (default: from GCP_PROJECT_ID env var)")
+    private String userProject = DownloadConstants.GCP_PROJECT_ID;
+
+    /**
+     * GCP mainnet bucket helper; assumed to have:
+     *  - listHour(LocalDate date, int hour, boolean includeSidecars)
+     */
+    private MainNetBucket mainNetBucket;
+
+    @Override
+    public void run() {
+        try {
+            if (mainNetBucket == null) {
+                mainNetBucket = new MainNetBucket(
+                        cacheEnabled, cacheDir.toPath(), minNodeAccountId, maxNodeAccountId, userProject);
+            }
+            if (!skipLocalCompare && downloadedDaysDir == null) {
+                throw new NotFoundException("DownloadedDaysDir has not been set");
```

Comment:
Just use an `IllegalArgumentException` here also as this only happens when `downloadedDaysDir` is invalid.
Seems too much to add helidon `NotFoundException` just for this